lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm

```diff
@@ -0,0 +1,280 @@
+genome model clin-seq identify-loh\\
+    --outdir=/tmp/test/ \\
+    --somvar-build='9dc0385a1b634c9bb85eb2017b3a0c73'
+EOS
+}
+
+sub help_detail {
+    return <<EOS
+Use the results from Varscan in the somatic-variation builds and identify
+regions of LOH. Specifically, look at heterozygous single nucleotide variants
+in the normal sample and compare with stretches of homozygosity in the tumor.
+
+If a clinseq build is supplied as input, the tool attempts to use the results
+from the WGS-somatic-variation build input, if these are unavailable the tool
+looks for WEx-somatic-variation build.
+
+Alternatively, instead of a clin-seq build you can supply a
+somatic-variation build as an input.
+
+Thanks to Chris Miller for working out the workflow.
+EOS
+}
+
+sub __errors__ {
+    my $self = shift;
+    my @errors = $self->SUPER::__errors__(@_);
+    unless (-e $self->outdir && -d $self->outdir) {
+        push @errors, UR::Object::Tag->create(
+            type => 'error',
+            properties => ['outdir'],
+            desc => "Outdir: " . $self->outdir .
+                " not found or not a directory",
+        );
+    }
+    return @errors;
+}
+
+#Get a somatic-variation build from input or from clinseq-build
+sub resolve_somvar {
+    my $self = shift;
+    my $somvar_build = $self->somvar_build;
+    unless($somvar_build) {
+        my $clinseq_build = $self->clinseq_build;
+        unless($clinseq_build) {
+            die $self->error_message("Please pass a somatic-variation
+                or clinseq build as input.");
+        }
+        $somvar_build = $self->get_best_somvar_build($clinseq_build);
+    }
+    return $somvar_build;
+}
+
+#Take Varscan SNVs and split into per-chromosome calls.
+sub split_snvs {
+    my $self = shift;
+    my $varscan_op = shift;
+    my $snv_prefix = shift;
+    my $awk_cmd;
+    #use subsample of chr1 variants for test
+    if($self->test) {
+        $awk_cmd = "awk '\$1==1 { if(NR\%1000 == 0) { print > \"$snv_prefix.\"\$1\".snp\" }}' $varscan_op";
+    } else {
+        $awk_cmd = "awk '{ print > \"$snv_prefix.\"\$1\".snp\" }' $varscan_op";
+    }
+    Genome::Sys->shellcmd(cmd => $awk_cmd);
+    return;
+}
+
+#Get Varscan SNVs from somatic-variation
+sub get_varscan_snvs {
+    my $self = shift;
+    my $somvar_build = shift;
+    my $snv_prefix = shift;
+    my $dd = $somvar_build->data_directory;
+    my @varscan_snvs =
+    glob(File::Spec->join($dd, "variants/snv/varscan-somatic-*/snvs.hq.unfiltered"));
+    unless(scalar @varscan_snvs) {
+        die $self->error_message("Unable to find varscan SNVs in $dd");
+    }
+    $self->status_message("varscan snvs file found: " .
+        join("\t", @varscan_snvs));
+    $self->split_snvs($varscan_snvs[0], $snv_prefix);
+}
+
+#Get tumor/normal BAMs from somatic-variation
+sub get_tumor_normal_bam {
+    my $self = shift;
+    my $somvar_build = shift;
+    my $tumor_bam = $somvar_build->tumor_model->last_succeeded_build->
```

Comment:
In general, when starting from a build, it is bad practice to jump to a model and thence to its last succeeded build.  There is a `tumor_build` on the `$somvar_build` that corresponds to the build actually used by the SomaticVariation build (which may be quite different from some later build).